internal/cmd/tui.go

```diff
@@ -133,7 +133,14 @@ func tuiCmd() *cobra.Command {
 						return err
 					}
 
-					err = runnerClient.RunBlock(ctx, result.block)
+					tempBlock := result.block.Block.Clone()
+
+					err = promptEnvVars(cmd, []project.FileCodeBlock{&tempBlock})
+					if err != nil {
+						return err
+					}
+
+					err = runnerClient.RunBlock(ctx, &tempBlock)
```

Comment:
Calling `RunBlock` directly on the block prevents the runner from determining the correct location for `cwd`, since `document.Block` has no indication of file location.

I think implementing a `Clone` method for `project.CodeBlock` and using that instead would fix this issue.